src/script/view_model/content/GroupCreationViewModel.ts

```diff
@@ -170,10 +182,42 @@ export class GroupCreationViewModel {
     this.isShown(false);
   };
 
+  readonly clickOnToggleServicesMode = (): void => {
+    let newAccessState: TEAM;
+
+    if (this.isGuestRoom()) {
+      newAccessState = ACCESS_STATE.TEAM.GUESTS_SERVICES;
+    }
+
+    if (this.isGuestAndServicesRoom()) {
+      newAccessState = ACCESS_STATE.TEAM.GUEST_ROOM;
+    }
+    if (this.isServicesRoom()) {
+      newAccessState = ACCESS_STATE.TEAM.TEAM_ONLY;
+    }
+    if (this.isTeamOnly()) {
+      newAccessState = ACCESS_STATE.TEAM.SERVICES;
+    }
+
+    this.accessState(newAccessState);
```

Comment:
Ok yeah, not too found of those, somewhat, duplicated lines, I think we can simplify this logic and have a generic helper that converts `ACCESS_STATE` to `ACCESS_ROLES_V2` and vice versa. 

Here we would do something like this. 

```
toggle<AccessRoleV2> {
  currentRoles = this.conversation.accessRolesV2() // ? not sure about where this is
  if (currentRoles.includes(TheRoleWeToggle)) {
    currentRoles.remove(ACCESS_ROLE_V2.ROLE);
  } else {
    currentRoles.push(ACCESS_ROLE_V2.ROLE);
  }

  // Compute the new accessState from the updated roles
  this.accessState(computeStateFromRoles(currentRoles);
}
```

I'd say this is non-blocker for this PR, but if you could investigate this and maybe open a follow up PR, that'd be great :)

Reply:
created new ticket for visibility https://wearezeta.atlassian.net/browse/SQSERVICES-1289